I checked https://s3.amazonaws.com/clickhouse-test-reports/70943/5cf74f4da2d71bf971d720c6a38bc00e00c3cf36/integration_tests__asan__old_analyzer__[1_6]//home/ubuntu/actions-runner/_work/_temp/test/output_dir/integration_run_parallel4_0.tar.zst


```
cat ./_instances-0-gw0/resolver/no_list_objects.err.log
```

```
127.0.0.1 - - [22/Oct/2024 19:26:44] "GET / HTTP/1.1" 200 -
172.16.5.7 - - [22/Oct/2024 19:28:52] "PUT /root/test_no_list_6.csv HTTP/1.1" 200 -
----------------------------------------
Exception occurred during processing of request from ('172.16.5.7', 57854)
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
172.16.5.7 - - [22/Oct/2024 19:28:52] "PUT /root/test_no_list_12.csv HTTP/1.1" 200 -
  File "/usr/local/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/local/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
  File "/usr/local/lib/python3.12/http/server.py", line 436, in handle
    self.handle_one_request()
  File "/usr/local/lib/python3.12/http/server.py", line 425, in handle_one_request
    self.wfile.flush() #actually send the response if not already done.
    ^^^^^^^^^^^^^^^^^^
ValueError: I/O operation on closed file.
----------------------------------------
----------------------------------------
Exception occurred during processing of request from ('172.16.5.7', 57868)
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/local/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/local/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
  File "/usr/local/lib/python3.12/http/server.py", line 436, in handle
    self.handle_one_request()
  File "/usr/local/lib/python3.12/http/server.py", line 425, in handle_one_request
    self.wfile.flush() #actually send the response if not already done.
    ^^^^^^^^^^^^^^^^^^
ValueError: I/O operation on closed file.
----------------------------------------
```

```
cat ./_instances-0-gw0/resolver/unstable_server.err.log
127.0.0.1 - - [22/Oct/2024 19:26:46] "GET / HTTP/1.1" 200 -
172.16.5.7 - - [22/Oct/2024 19:30:45] "HEAD /root/slow_send_test.csv HTTP/1.1" 200 -
172.16.5.7 - - [22/Oct/2024 19:30:45] "GET /root/slow_send_test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:30:46] "GET /root/slow_send_test.csv HTTP/1.1" 206 -
----------------------------------------
Exception occurred during processing of request from ('172.16.5.7', 39766)
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/local/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/local/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
  File "/usr/local/lib/python3.12/http/server.py", line 436, in handle
    self.handle_one_request()
  File "/usr/local/lib/python3.12/http/server.py", line 424, in handle_one_request
    method()
  File "//unstable_server.py", line 111, in do_GET
    self.wfile.write(
  File "/usr/local/lib/python3.12/socketserver.py", line 840, in write
    self._sock.sendall(b)
BrokenPipeError: [Errno 32] Broken pipe
----------------------------------------
172.16.5.7 - - [22/Oct/2024 19:34:10] "HEAD /root/test.csv HTTP/1.1" 200 -
172.16.5.7 - - [22/Oct/2024 19:34:10] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:10] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:10] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:10] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:11] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:12] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:12] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:12] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:12] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:12] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:13] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:13] "GET /root/test.csv HTTP/1.1" 206 -
172.16.5.7 - - [22/Oct/2024 19:34:13] "GET /root/test.csv HTTP/1.1" 206 -
```

in test_storage_s3 for example.

--- a/tests/integration/test_storage_s3/s3_mocks/no_list_objects.py
+++ b/tests/integration/test_storage_s3/s3_mocks/no_list_objects.py
@@ -110,7 +110,6 @@ class RequestHandler(http.server.BaseHTTPRequestHandler):
             self.send_header(k, v)
         self.end_headers()
         self.wfile.write(r.content)
-        self.wfile.close()
 
 
 class ThreadedHTTPServer(socketserver.ThreadingMixIn, http.server.HTTPServer):

